strawberry/types/type_resolver.py

```diff
@@ -50,6 +51,15 @@ class if one is not set by either using an explicit strawberry.field(name=...) o
     # Deferred import to avoid import cycles
     from strawberry.field import StrawberryField
 
+    # Relay uses @interface/@type while creating its types, which calls this function
+    try:
+        from strawberry.relay import Connection, ConnectionField, Node, NodeField
+    except ImportError:
+        Connection = None  # type: ignore[misc,assignment]
+        ConnectionField = None  # type: ignore[misc,assignment]
+        Node = None  # type: ignore[misc,assignment]
+        NodeField = None  # type: ignore[misc,assignment]
```

Comment:
why would we get an import error here?

Reply:
Inside `relay/types.py` the types defined there use `@strawberry.type`, which imports this file to resolve their fields and causes a circular import error.

Probably my comment above is not enough to explain the issue here =P. But when creating the types there, `Node`, `Edge` and `Connection`, this will raise an `ImportError`. After that the import should work fine

Reply:
this looks like something we need to refactor, either in this PR or another one :D

Reply:
I refactored it to make the code better to understand